### PR TITLE
Rebuild index.js on Dependabot PR.

### PR DESCRIPTION
**Pull request by figi44**
Adds a workflow to rebuild dist/index.js when Dependabot opens a pull request.

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](http://claassistant.ecmwf.int:80/ecmwf-actions/build-package?pullRequest=58) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](http://claassistant.ecmwf.int:80/ecmwf-actions/build-package?pullRequest=58) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](http://claassistant.ecmwf.int:80/check/ecmwf-actions/build-package?pullRequest=58) it.</sub>

File: .github/workflows/dependabot-build.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run linter
+        run: npm run lint-no-fix
```
Comment (dtip) [inline .github/workflows/dependabot-build.yml:32]: Can probably skip this

File: .github/workflows/dependabot-build.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: npm run build
+
+      - name: Run tests
+        run: npm test
+
+      - name: push index.js
+        run: |
+          git config --global user.email "dependabot@example.com"
+          git config --global user.name "dependabot"
+          git checkout -b ${{github.head_ref}}
```
Comment (dtip) [inline .github/workflows/dependabot-build.yml:44]:
```suggestion
          git checkout -b ${{ github.head_ref }}
```
Idiomatic GitHub Actions YAML syntax typically has spaces around the expression in the curly brackets.

**Review by dtip — CHANGES_REQUESTED** [review]
This is a great start - reckon it needs a few tweaks before it'll be bulletproof!

File: .github/workflows/dependabot-build.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+# permissions needed for git push
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+    name: Build
+    if: github.actor == 'dependabot[bot]'
```
[the remark this answers is not in the document]
Comment (dtip) [reply .github/workflows/dependabot-build.yml:19]: Might make more sense to use the `paths` filter to see if dependabot has changed `package.json`

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

File: .github/workflows/dependabot-build.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# This workflow rebuilds dist/index.js on opened pull request by Dependabot and adds it to the PR
+name: dependabot-build
+
+# trigger workflow only on opened PRs or on demand
+on: 
+  pull_request:
```
[the remark this answers is not in the document]
Comment (figi44) [reply .github/workflows/dependabot-build.yml:6]: `pull_request` should have enough permissions to push a commit, I tested it separately and it worked.

**Review by dtip — APPROVED** [review]
👍

**Pull request merged**